src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -1015,7 +1015,12 @@ foam.CLASS({
       code: function(X, memo) {
         var approvedApprovalRequest = this.clone();
         approvedApprovalRequest.status = this.ApprovalStatus.APPROVED;
-        approvedApprovalRequest.memo = memo;
+        approvedApprovalRequest.memo = memo + ' -- '
+                                        + X.subject.user.firstName + ' '
+                                        + X.subject.user.lastName
+                                        + ' ' + (new Date()).toString()
+                                        + (approvedApprovalRequest.memo && '\n')
+                                        + approvedApprovalRequest.memo;
 
```

Comment:
can we turn this into a function where you pass context and and the object with the memo?